py/fiberassign/utils.py

```diff
@@ -59,3 +59,76 @@ def option_list(opts):
                 else:
                     optlist.append("{}".format(val))
     return optlist
+
+# This is effectively the "Painter's Partition Problem".
+
+def distribute_required_groups(A, max_per_group):
+    ngroup = 1
```

Comment:
Please include a docstring for `distribute_required_groups` and `distribute_partition`